src/toil/leader.py

```diff
@@ -1160,78 +1227,73 @@ def processFinishedJob(self, batchSystemID, result_status, wallTime=None, exitRe
-                self.processRemovedJob(issuedJob, result_status)
-                return
-            if replacementJob.logJobStoreFileID is not None:
-                with replacementJob.getLogFileHandle(self.jobStore) as logFileStream:
+                'the consistency of your job store storage!', finished_job)
+                self.processRemovedJob(finished_job, result_status)
+                return False
+            if replacement_job.logJobStoreFileID is not None:
+                with replacement_job.getLogFileHandle(self.jobStore) as log_stream:
                     # more memory efficient than read().striplines() while leaving off the
                     # trailing \n left when using readlines()
                     # http://stackoverflow.com/a/15233739
-                    StatsAndLogging.logWithFormatting(jobStoreID, logFileStream, method=logger.warning,
-                                                      message='The job seems to have left a log file, indicating failure: %s' % replacementJob)
+                    StatsAndLogging.logWithFormatting(job_store_id, log_stream, method=logger.warning,
+                                                      message='The job seems to have left a log file, indicating failure: %s' % replacement_job)
                 if self.config.writeLogs or self.config.writeLogsGzip:
-                    with replacementJob.getLogFileHandle(self.jobStore) as logFileStream:
-                        StatsAndLogging.writeLogFiles(replacementJob.chainedJobs, logFileStream, self.config, failed=True)
+                    with replacement_job.getLogFileHandle(self.jobStore) as log_stream:
+                        StatsAndLogging.writeLogFiles(replacement_job.chainedJobs, log_stream, self.config, failed=True)
             if result_status != 0:
                 # If the batch system returned a non-zero exit code then the worker
                 # is assumed not to have captured the failure of the job, so we
                 # reduce the try count here.
-                if replacementJob.logJobStoreFileID is None:
-                    logger.warning("No log file is present, despite job failing: %s", replacementJob)
-
-                # Look for any standard output/error files created by the batch system.
-                # They will only appear if the batch system actually supports
-                # returning logs to the machine that submitted jobs, or if
-                # --workDir / TOIL_WORKDIR is on a shared file system.
-                # They live directly in the Toil work directory because that is
-                # guaranteed to exist on the leader and workers.
-                workDir = Toil.getToilWorkDir(self.config.workDir)
-                # This must match the format in AbstractBatchSystem.formatStdOutErrPath()
-                batchSystemFilePrefix =  f'toil_{self.config.workflowID}.{batchSystemID}'
-                batchSystemFileGlob = os.path.join(workDir, batchSystemFilePrefix + '*.log')
```

Comment:
It's good to see that you made `format_std_out_err_glob()` that is next to `formatStdOutErrPath()`!